models/solution/route.rb

```diff
@@ -55,6 +55,34 @@ def insert_stop(vrp, stop, index, idle_time = 0)
         shift_route_times(idle_time + stop.activity.duration, index)
       end
 
+      def remove_stop(vrp, stop, index)
+        removed_stop = stops.delete_at(index)
+        if index != 0 && index != stops.size
+          previous_stop = stops[index - 1]
+          next_stop = stops[index]
+
+          next_start = next_stop.info.begin_time
+          next_matrix_index = stops[index..-1].find{ |n_stop| n_stop.activity.point }&.activity&.point&.matrix_index
```

Comment:
Why can't we do following?
```suggestion
          next_matrix_index = next_stop.activity.point.matrix_index
```

Reply:
The next may be a rest which have no point :/

Reply:
:disappointed_relieved: bien-vu